torch/csrc/jit/ir.h

```diff
@@ -152,6 +157,23 @@ struct Node : public Attributes<Node> {
   const std::string & debugName() const {
     return debug_name_;
   }
+  const std::vector<Symbol> scope() const {
+    return scope_;
+  }
+  std::string scopeName() const {
+    std::string scope_name;
+    if (scope_.empty()) {
+      return scope_name;
+    }
+    scope_name = std::accumulate(
```

Comment:
Apologies, but looking at this part of the patch I am now not sure about the wisdom of making these interned strings. The original motivation for interning was to make it easier to test for equality on scopes, but it looks like this isn't actually being used ever, so maybe we shouldn't intern here. I wonder if @zdevito has thoughts.

Reply:
I was thinking about this. The advantage right now is that it might be lighter if you have a lot of nodes, but you're right, we're not matching at the moment.
Could we ever expose interned strings to python without converting to strings?

I'm fine reverting to strings anyway.

Reply:
I'd wait and see if anyone has opinions. If not, let's perhaps not bother changing it; we're allocating a string per scope and there should only be finitely many of these.

Reply:
Interning makes sense, because without it every Node would keep a copy of names of all parent scopes, which is quite a lot. However, if someone uses varying scope names this can really pollute the interned pool. I think it would be nicer to implement scoping mechanism in a trie-like approach, where scopes build up a tree (with `shared_ptr`s to parent nodes) with names at each level, and `Node`s only point to their node in the scope tree.